### DISCUSSION TITLE
RESEARCH: boot from USB drive

### DISCUSSION THREAD

**Discussion by fluidvoice** [raspiblitz/raspiblitz#3947] [Research]
Enhancement:  This is just to doc my investigation into how to be able to boot RaspiBlitz from the USB drive. Ideally to choose to boot from either SD-card or HDD via a script or boot loader at startup time. There are various reasons why you might want to do this.  Have two RaspiBlitz versions installed for testing (one working backup)... have one full version of Raspbian installed for use with an HDMI monitor as full Linux computer... etc.
I plugged my 500GB HDD into another Linux machine and used Gparted to shrink the BLOCKCHAIN partition **sda1** to 341GB and created a boot partition as **sda2**, a 120GB partition for Raspbian as  **sda3**, and a 2GB swap partition as **sda4** just if it might be needed later.
I restarted and Raspiblitz is still working and syncing so the blockchain DB was not corrupted.
Now I need to figure out how to `rsync` the SD card data over to the 120GB partition and make it active/bootable. 

$ lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda           8:0    0 465.8G  0 disk 
├─sda1        8:1    0 341.2G  0 part /mnt/hdd
├─sda2        8:2    0  1000M  0 part /mnt/target/boot
├─sda3        8:3    0 121.7G  0 part /mnt/target
└─sda4        8:4    0     2G  0 part 
mmcblk0     179:0    0  29.8G  0 disk 
├─mmcblk0p1 179:1    0  43.8M  0 part /boot
└─mmcblk0p2 179:2    0  29.8G  0 part /

**Comment by 21isenough** [comment]
I'm running a RaspiBolt on a Pi3 and got rid of the SD card but instead use an external USB stick to boot from ((1) external HDD with blockchain and (2) USB mass storage with boot partition).

This will only work on the Pi 3 since the boot code is stored in the BCM2837 device only, so the Pi1, Pi2, and Pi Zero will all require SD cards. However, you can boot a Pi1 and Pi2 using a special SD card that only contains the single file bootcode.bin. 

In order to achieve booting from a mass storage device, the USB boot bit needs to be set in the OTP (one-time programmmble) memory. This is done by booting from an SD card with a config option to enable USB boot mode. In my case this wasn't enough: The timing during the boot sequence it crucial. I needed to increase the timeout in order to give the usb stick enough time to "get ready". 

Those two pages should contain all the information needed:
https://www.raspberrypi.org/documentation/hardware/raspberrypi/bootmodes/msd.md
https://www.raspberrypi.org/blog/pi-3-booting-part-i-usb-mass-storage-boot/
